test/rails_test.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class TestRails < Minitest::Test
+  ERROR_TIMEOUT = 5
+  ERROR_THRESHOLD = 1
+  SEMIAN_OPTIONS = {
+    name: :testing,
+    tickets: 1,
+    timeout: 0,
+    error_threshold: ERROR_THRESHOLD,
+    success_threshold: 2,
+    error_timeout: ERROR_TIMEOUT,
+  }
+
+  def setup
+    @config = {
+      adapter: "mysql2",
+      connect_timeout: 2,
+      read_timeout: 2,
+      write_timeout: 2,
+      reconnect: true,
+      prepared_statements: false,
+      host: SemianConfig["toxiproxy_upstream_host"],
+      port: SemianConfig["mysql_toxiproxy_port"],
+    }
+
+    ActiveRecord::Base.establish_connection(@config)
+
+    @connection = ActiveRecord::Base.connection
+  end
+
+  def test_connection_has_a_semian_resource
+    assert(@connection.semian_resource, "expected semian_resource to be available on @connection")
+  end
+
+  def test_semian_resource_is_available_after_disconnect
+    assert(@connection.semian_resource, "expected semian_resource to be available on @connection")
+
+    @connection.disconnect!
+
+    assert(@connection.semian_resource, "expected semian_resource to be available on @connection")
+  end
+
+  def test_semian_resource_is_available_after_reset
+    assert(@connection.semian_resource, "expected semian_resource to be available on @connection")
+
+    @connection.reset!
+
+    assert(@connection.semian_resource, "expected semian_resource to be available on @connection")
+  end
+
+  def test_semian_resource_is_available_after_reconnect
+    assert(@connection.semian_resource, "expected semian_resource to be available on @connection")
+
+    @connection.reconnect!
+
+    assert(@connection.semian_resource, "expected semian_resource to be available on @connection")
+  end
+end
```

Comment:
Can you also add test to check the result of `semian_resource`?
I would like to be sure that config is not just `nil`, 
but is equal to `SEMIAN_OPTIONS`